### PR TITLE
do not ignore directories starting with . when scanning for projects

### PR DESCRIPTION
**Pull request by vladak**
This change allows projects to have any name that can be represented on the file system.

### CODE REVIEW

**Comment by idodeclare** [comment]
As I wrote on the linked issue, I'm not sure it was correctly diagnosed, so I recommend holding off on this change.

File: opengrok-indexer/src/main/java/org/opengrok/indexer/index/Indexer.java [hunk 1]
```diff
@@ -946,7 +946,7 @@ public void prepareIndexer(RuntimeEnvironment env,
                     p.setName(name);
                     p.completeWithDefaults();
                     projects.put(name, p);
-                } else if (!name.startsWith(".") && file.isDirectory()) {
+                } else if (file.isDirectory()) {
```
Comment (tulinkry) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/index/Indexer.java]: what about `.` and `..` ?

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/Indexer.java]: https://docs.oracle.com/javase/7/docs/api/java/io/File.html#listFiles() says:
> an array of File objects is returned, one for each file or directory in the directory. Pathnames denoting the directory itself and the directory's parent directory are not included in the result.

**Comment by vladak** [comment]
> As I wrote on the linked issue, I'm not sure it was correctly diagnosed, so I recommend holding off on this change.

removed the issue reference in the changeset.

**Comment by vladak** [comment]
I hope this is non-controversial so could be merged.

**Comment by idodeclare** [comment]
It's controversial because the handling is a deliberate and because it seems the genesis of this patch was a misdiagnosed issue. Who truly desires this to be removed? Could some user be broken for no good benefit?

**Comment by vladak** [comment]
Okay, I don't feel strongly about this so closing.

**Pull request closed without merging**